src/main/java/org/nuxeo/segment/io/SegmentIOComponent.java

```diff
@@ -185,97 +181,102 @@ public void identify(NuxeoPrincipal principal) {
     }
 
     @Override
-    public Providers getProviders() {
-        if (providers == null) {
-            providers = new Providers();
-            if (providersConfig != null) {
-                if (!providersConfig.enableDefaults) {
-                    providers.setDefault(false);
-                }
-                for (String name : providersConfig.providers.keySet()) {
-                    providers.setEnabled(name,
-                            providersConfig.providers.get(name));
-                }
-            }
+    public Map<String, Boolean> getIntegrations() {
+        if (integrationsConfig != null && integrationsConfig.integrations != null) {
+            return integrationsConfig.integrations;
+        }
+        return new HashMap<>();
+    }
+
+    protected Options buildOptions() {
+        Options options = new Options();
+        for (Entry<String, Boolean> integration : getIntegrations().entrySet()) {
+            options.setIntegration(integration.getKey(), integration.getValue());
         }
-        return providers;
+        return options.setTimestamp(new DateTime());
     }
 
     @Override
-    public void identify(NuxeoPrincipal principal,
-            Map<String, Serializable> metadata) {
+    public void identify(NuxeoPrincipal principal, Map<String, Serializable> metadata) {
 
-        SegmentIODataWrapper wrapper = new SegmentIODataWrapper(principal,
-                metadata);
+        SegmentIODataWrapper wrapper = new SegmentIODataWrapper(principal, metadata);
 
         if (!mustTrackprincipal(wrapper.getUserId())) {
             log.debug("Skip user " + principal.getName());
             return;
         }
 
-        if (Framework.isTestModeSet()) {
-            pushForTest("identify", wrapper.getUserId(), null, metadata);
+        if (debugMode) {
```

Comment:
no, the testMode part has just been moved just beside the "real" API code to allow calling it with exactly the same arguments